components/layout_2020/table/layout.rs

```diff
@@ -799,7 +799,7 @@ impl<'a> TableLayout<'a> {
     /// Distribute width to columns, performing step 2.4 of table layout from
     /// <https://drafts.csswg.org/css-tables/#table-layout-algorithm>.
     fn distribute_width_to_columns(&self) -> Vec<Au> {
-        if self.table.slots.is_empty() {
+        if self.table.size.width.is_zero() {
```

Comment:
It might be worth a comment here to say that a table might have columns, but no rows.